src/orchestrator.ts

```diff
@@ -167,11 +159,14 @@ export class Orchestrator {
                     return;
                 }
 
-                const newDecisionStartedEvent = state.find(
-                    (e) =>
-                        e.EventType === HistoryEventType.OrchestratorStarted &&
-                        e.Timestamp > decisionStartedEvent.Timestamp
-                );
+                const newDecisionStartedEvent = state
+                    .filter(
+                        (e, index) =>
+                            e.EventType === HistoryEventType.OrchestratorStarted &&

```

Comment:
I am like 90% sure this is correct (or at least as correct as we can be with our current per-Task linear scan algorithm).

I am a bit uncertain about the case where we have a `df.task.whenAll()` with a bunch of `callActivityWithRetry()` operations that may finish out of order due to retries, but I think the `e.Timestamp > decsionStartedEvent.Timestamp` will help prevent us from traveling "backwards" in time as we process subsequent retried activities that may have finished earlier.

In that case, this filtered list should be empty, and `.pop()` will just return undefined and we will keep the current value for `decisionStartedEvent`.